File: client/src/types/reportExtraction.ts
export type OcrStatus = "processing" | "completed" | "failed";

export interface MedicalParameter {
  parameter: string;
  value: string;
  unit: string | null;
  reference_range: string | null;
}

export interface ReportExtraction {
  id: string;
  report_id: string;
  raw_text: string | null;
  structured_json: MedicalParameter[] | null;
  ocr_status: OcrStatus;
  processing_started_at: string | null;
  completed_at: string | null;
  created_at: string;
}

File: server/app/models/report_extraction.py
from sqlalchemy import Column, DateTime, ForeignKey, String, Text, text
from sqlalchemy.dialects.postgresql import JSONB, UUID

from app.db.session import Base


class ReportExtraction(Base):
    __tablename__ = "report_extractions"

    id = Column(UUID(as_uuid=True), primary_key=True, server_default=text("gen_random_uuid()"))
    report_id = Column(
        UUID(as_uuid=True),
        ForeignKey("reports.id", ondelete="CASCADE"),
        nullable=False,
        unique=True,
        index=True,
    )
    raw_text = Column(Text, nullable=True)
    structured_json = Column(JSONB, nullable=True)
    ocr_status = Column(String, nullable=False, server_default="processing")
    processing_started_at = Column(DateTime(timezone=True), nullable=True)
    completed_at = Column(DateTime(timezone=True), nullable=True)
    created_at = Column(DateTime(timezone=True), server_default=text("now()"))